openid-4-verifiable-credential-issuance-1_0.md

```diff
@@ -1144,6 +1146,96 @@ Cache-Control: no-store
+
+Communication with the Acknowledgement Endpoint MUST utilize TLS.
+
+## Acknowledgement Request {#acknowledgement}
+
+The Wallet sends an HTTP POST request to the Acknowledgement Endpoint with the following parameters in the entity-body and using the `application/json` media type.
+
+* `credentials`: Array of objects, where each object consists of the following parameters:
+  * `ack_id`: REQUIRED. String received in Credential Response or Batch Credential Response.
+  * `status`: REQUIRED. Status whether the credential issuance was successful or not. It MUST be a case sensitive string whose value is either `success`, `failure` or `rejected`. `rejected` is to be used when the unsuccessful credential issuance was caused by a user action. In all other unsuccessful cases, `failure` is to be used.
+  * `error_description`: OPTIONAL. Human-readable ASCII [@!USASCII] text providing additional information, used to assist the Credential Issuer developer in understanding the error that occurred. Values for the `error_description` parameter MUST NOT include characters outside the set `%x20-21 / %x23-5B / %x5D-7E`.
+
+Below is a non-normative example of an acknowledgement request when credential issuance was successful:
+
+```
+POST /acknowledgement HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Authorization: Bearer czZCaGRSa3F0MzpnWDFmQmF0M2JW
+
+{
+  "credentials": [
+    {
+    "ack_id": "3fwe98js",
+    "status": "success"
+    }
+  ]
+}
+```
+
+Below is a non-normative example of an acknowledgement request when credential issuance was unsuccessful:
+
+```
+POST /acknowledgement HTTP/1.1
+Host: server.example.com
+Content-Type: application/json
+Authorization: Bearer czZCaGRSa3F0MzpnWDFmQmF0M2JW
+
+{
+  "credentials": [
+    {
+    "ack_id": "3fwe98js",
+    "status": "failure",
+    "error_description": "..."
+    }
+  ]
+}
+```
+
+## Successful Acknowledgement Response
+
+When the Credential Issuer has successfully received the acknowledgement request from the Wallet, it MUST respond with the HTTP status code 2xx. The usage of the HTTP status code 204 (No Content) is RECOMMENDED.
+
+Below is a non-normative example of response to a successful acknowledgement request:
+
+```
+HTTP/1.1 204 No Content
+```
+
+## Acknowledgement Error Response
+
+If the acknowledgement request does not contain an Access Token or contains an invalid Access Token, the Acknowledgement Endpoint returns an authorization error response such as defined in section 3 of [@!RFC6750].
+
+`invalid_request` parameter defined in section 3.1 of [@!RFC6750] SHOULD be used in most cases other than when `ack_id` value is invalid. In the latter case, the following error code SHOULD be used as the value of the `error` parameter:
```

Comment:
RFC6750 defines use of the WWW-Authenticate response header field for the error param. Which isn't really the appropriate place for application layer errors like an invalid ack_id. And also isn't what's shown in the example Acknowledgement Error Response.

Reply:
good catch - fixed to 400 application/json requirements

Reply:
Still isn't quite right.